node/src/reactor/validator.rs

```diff
@@ -736,6 +736,86 @@ impl reactor::Reactor for Reactor {
                             warn!("received get request for gossiped-address from {}", sender);
                             return Effects::new();
                         }
+                        Tag::BlockHeaderByHash => {
+                            let block_hash: BlockHash = match bincode::deserialize(&serialized_id) {
+                                Ok(block_hash) => block_hash,
+                                Err(error) => {
+                                    error!(
+                                        "failed to decode {:?} from {}: {}",
+                                        serialized_id, sender, error
+                                    );
+                                    return Effects::new();
+                                }
+                            };
+
+                            match self.storage.read_block_header_by_hash(&block_hash) {
+                                Ok(Some(block_header)) => {
+                                    match Message::new_get_response(&block_header) {
+                                        Ok(message) => {
+                                            return effect_builder
+                                                .send_message(sender, message)
+                                                .ignore();
+                                        }
+                                        Err(error) => {
+                                            error!("failed to create get-response: {}", error);
+                                            return Effects::new();
+                                        }
+                                    };
+                                }
+                                Ok(None) => {
+                                    debug!("failed to get {} for {}", block_hash, sender);
+                                    return Effects::new();
+                                }
```

Comment:
Not in this PR but I think we should be signaling to the caller that we don't have the item, instead of letting the peer time out. I think it will have a performance improvement on the network.